js/views/scrollView.js

```diff
@@ -772,7 +773,21 @@ ionic.views.Scroll = ionic.views.View.inherit({
         }
       }
 
-      self.doTouchMove(getEventTouches(e), e.timeStamp, e.scale);
+      if(self.options.locking && self.__scrollLockFix.direction === null){
+        var detect = 5,
+            dx = Math.abs(e.touches[0].pageX - self.__initialTouchLeft), dy = Math.abs(e.touches[0].pageY - self.__initialTouchTop);
+        if(dx > detect || dy > detect){
+          self.__scrollLockFix.direction = dx > dy;
+          self.__scrollLockFix.doFix = true;
+        }
+      }
+      var dir = self.__scrollLockFix.direction;
+      if(self.__scrollLockFix.doFix){
+        self.doTouchMove([{pageX: dir ? e.touches[0].pageX : self.__initialTouchLeft, pageY: dir ? self.__initialTouchTop : e.touches[0].pageY}], e.timeStamp, e.scale);
```

Comment:
This line is too long. I would refactor it into something like below (but it's obviously a matter of a personal preference :smiley:): 

```js
var coords = dir ?
  {pageX: e.touches[0].pageX, pageY: self.__initialTouchTop} :
  {pageX: self.__initialTouchLeft, pageY: e.touches[0].pageY};
self.doTouchMove([coords], e.timeStamp, e.scale);
```